Jellyfin/Views/VideoViewModel.cs

```diff
@@ -70,12 +81,134 @@ public async void HandleParameters(Video.Parameters parameters)
+                            SubtitleStreamIndex = parameters.SubtitleStreamIndex,
+                            PlaySessionId = _playingSessionId,
+                            PositionTicks = await GetCurrentTicks(),
+                            SessionId = _appSettings.SessionId,
+                            IsPaused = false
+                        });
+                        break;
+                }
+            };
+
+            await _jellyfinApiClient.PlayingItems[videoId].PostAsync(request =>
+            {
+                request.QueryParameters.MediaSourceId = videoId.ToString("N");
+                request.QueryParameters.AudioStreamIndex = parameters.AudioStreamIndex;
+                request.QueryParameters.SubtitleStreamIndex = parameters.SubtitleStreamIndex;
+                request.QueryParameters.PlaySessionId = _playingSessionId = playbackInfo.PlaySessionId;
+                // TODO: do we need to support sessions/sessionId?
+                request.QueryParameters.CanSeek = _playerElement.MediaPlayer.PlaybackSession.CanSeek;
+
+            });
+
+            _progressTimer = new Timer(async (o) =>
+            {
+                if (await IsPlaying())
+                {
+                    Int64 currentTicks = await GetCurrentTicks();
+                    _ = _jellyfinApiClient.PlayingItems[videoId].Progress.PostAsync(request =>
+                    {
+                        request.QueryParameters.MediaSourceId = videoId.ToString("N");
+                        request.QueryParameters.AudioStreamIndex = parameters.AudioStreamIndex;
+                        request.QueryParameters.SubtitleStreamIndex = parameters.SubtitleStreamIndex;
+                        request.QueryParameters.PlaySessionId = _playingSessionId = playbackInfo.PlaySessionId;
+                        request.QueryParameters.PositionTicks = currentTicks;
+
+                    });
+                    _ = _jellyfinApiClient.Sessions.Playing.Progress.PostAsync(new PlaybackProgressInfo()
+                    {
+                        CanSeek = true,
+                        ItemId = videoId,
+                        MediaSourceId = videoId.ToString("N"),
+                        AudioStreamIndex = parameters.AudioStreamIndex,
+                        SubtitleStreamIndex = parameters.SubtitleStreamIndex,
+                        PlaySessionId = _playingSessionId,
+                        PositionTicks = await GetCurrentTicks(),
+                        SessionId = _appSettings.SessionId
+                    });
+                }
+            }, videoId, TimeSpan.FromMinutes(1), TimeSpan.FromMinutes(1));
         }
         else
         {
             // Handle failure to create the adaptive media source
         }
     }
+
+    private async Task<Int64> GetCurrentTicks()
+    {
+        Int64 currentTicks = 0;
+        await Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
```

Comment:
(unsure) does this need to be run by the dispatcher? I'm not familiar enough with UWP to know tbh